src/pal/src/exception/signal.cpp

```diff
@@ -832,13 +846,9 @@ void handle_signal(int signal_id, SIGFUNC sigfunc, struct sigaction *previousAct
     struct sigaction newAction;
 
     newAction.sa_flags = SA_RESTART | additionalFlags;
-#if HAVE_SIGINFO_T
     newAction.sa_handler = NULL;
     newAction.sa_sigaction = sigfunc;
     newAction.sa_flags |= SA_SIGINFO;
-#else   /* HAVE_SIGINFO_T */
-    newAction.sa_handler = SIG_DFL;
```

Comment:
It was there as a fallback to enable at least some level of functionality on platforms where siginfo_t is not present. We would not be able to handle signals (which include hardware exceptions), but otherwise .NET core would work. This is a historical stuff from the Rotor time. But I think we can leave it there in case someone would like to try to bringup .NET on some exotic Unix.

Reply:
Is the assignment to SIG_DFL intentional? I thought it was an error.

Reply:
siginfo_t is POSIX 1993.

Reply:
I can see that Linux supports SA_SIGINFO since version 2.2 that was released in August 2000. So I guess you are right and we can just get rid of the check. But in that case, could you please remove it everywhere (including the cmake code that detects it)?